altair/vegalite/v5/api.py

```diff
@@ -345,6 +361,260 @@ def check_fields_and_encodings(parameter: Parameter, field_name: str) -> bool:
     return False
 
 
+# -------------------------------------------------------------------------
+# Tools for working with conditions
+_TestPredicateType = Union[str, _expr_core.Expression, core.PredicateComposition]
+"""TODO
+
+Item [(2)](https://vega.github.io/vega-lite/docs/condition.html) Specifying a `test` predicate: ...
+"""
+
+_PredicateType = Union[Parameter, core.Expr, TypingDict[str, Any], _TestPredicateType]
```

Comment:
`core.Expr` was in the union of `alt.condition(predicate)`, but the internal logic would raise a `NotImplementedError.
Keeping it around for now, but either the annotation or internals should be updated.